### PR TITLE
add long options

### PR DESCRIPTION
**Pull request by ThePiGrepper**
- fix issue #13

### CODE REVIEW

**Comment by falconindy** [comment]
You'll want to update the manpage and the shell completion scripts as well.

File: asp.in [hunk 1]
```diff
@@ -338,8 +338,25 @@ dispatch_action() {
 
 initialize || log_fatal 'failed to initialize asp repository in %s' "$ASPROOT"
 
-while getopts ':a:fhV' flag; do
+while getopts ':a:fhV-:' flag; do
```
Comment (falconindy) [inline asp.in]: This is kind of terrible. It means that --arch i686 isn't actually valid, like every other GNU getopts program.

Comment (ThePiGrepper) [reply asp.in]: yeah, makes. sense. will fix.

Comment (ThePiGrepper) [reply asp.in]: then which ones would be valid syntax?  --arch=<arch> and / or --arch <arch> ? , and what to do with --arch= <arch>? or --arch =<arch>?

File: asp.in [hunk 2]
```diff
@@ -338,8 +338,33 @@ dispatch_action() {
...
   case $flag in
+    -)
+      case "${OPTARG}" in
+        arch=*)
+          [[ -n "${OPTARG#*=}" ]] && OPT_ARCH="${OPTARG#*=}"
```
Comment (falconindy) [inline asp.in]: You need to fail when `--arch=` is passed.

File: asp.in [hunk 3]
```diff
@@ -338,8 +338,33 @@ dispatch_action() {
...
+          ;;
+        arch)
+          if [[ -n "${!OPTIND}" ]] ; then
+            OPT_ARCH="${!OPTIND}"
+            OPTIND=$(( $OPTIND + 1 ))
```
Comment (falconindy) [inline asp.in]: This should be written as `(( ++OPTIND ))`.

File: asp.in [hunk 4]
```diff
@@ -338,8 +338,33 @@ dispatch_action() {
...
+          ;;
+        help)
+          usage
+          exit 0
+          ;;
+        version)
+          version
+          exit 0
+          ;;
+      esac;;
```
Comment (falconindy) [inline asp.in]: newline between esac and ;;

And, what about the default case? We shouldn't just silently ignore `--ponies`

**Review by falconindy — COMMENTED** [review]
I kind of wonder if this is really worthwhile, or if we shouldn't just support --help/--version as long options. For a program that doesn't normally use options at all, I'm not convinced the added complexity is worthwhile.

**Comment by E5ten** [comment]
But with the new change --arch=i686 doesn't work now, and you can also do stuff like `-f-arch i686`  or `-f -arch i686` (an issue with anything that tries to use getopts for longopts).

**Comment by ThePiGrepper** [comment]
well, regarding the first point, at first, --arch=i686 was enabled and I was told to disable it, so that's that. regarding the second point, hmm, well that's a inherent problem with getopts as you say, it would be nice to have something better available

**Comment by falconindy** [comment]
I didn't mean to suggest that `--arch=i686` was invalid. Quite the opposite. It should be identical to `--arch i686`.

I'm really not seeing the point in doing this. I think the simpler solution is to:

1) Remove `-f`. It's used in only one place only (export), and it's usage is inconsistent, i.e. it could be used elsewhere but isn't. The value seems limited
2) Add explicit handling of --version and --help.
3) Tell people to suck it up about `-a` which is almost never needed.

**Comment by ThePiGrepper** [comment]
let's see:
1. ok, I could do that if you want, on a different pull probably. it looks like a 5min task.
2. when you say explicit, you mean out of the getopts and treated as another 'command'?
3. that makes sense, however, reading the issue, it looks like he is explicitly wanting that. ofc, it's your call.

**Comment by ThePiGrepper** [comment]
oh, and originally, --arch=i686 was identical to --arch i686. maybe I didnt quite understand your original request about removing it.

**Comment by E5ten** [comment]
@ThePiGrepper I think he meant you need to fail when --arch= alone was passed, not --arch=anything.

**Comment by ThePiGrepper** [comment]
that was already the case before I deleted that section.

**Pull request closed without merging**